### PR TITLE
fix(date-picker): focus on current date on down key

### LINKED ISSUE #3638: [DatePicker] React calendar control not keyboard accessible
When calendar control (React version) is focused, it is not able to navigate to specific dates using Keyboard keys. Issue can be recreated at http://react.carbondesignsystem.com/?path=/story/datepicker--single-with-calendar after pressing down arrow key to focus the calendar.

The Carbon calendar control (non-react) works as expected. https://www.carbondesignsystem.com/components/date-picker/code

## Environment

> Operating system
Mac OS

> Browser
Chrome
Mozilla

### PR DESCRIPTION
**Pull request by asudoh**
This change makes date picker focus on the currently selected date (if available) rather than the whole date picker dropdown, which allows Flatpickr code to work on keyboard nav with left/right keys after.

Fixes #3638.

#### Changelog

**Changed**

- A change to make date picker focus on the currently selected date (if available) rather than the whole date picker dropdown.

#### Testing / Reviewing

Testing should make sure out date picker is not broken.

### CODE REVIEW

File: packages/components/src/components/date-picker/date-picker.js [hunk 1]
```diff
@@ -142,7 +142,11 @@ class DatePicker extends mixin(
       this.manage(
         on(this.element, 'keydown', e => {
           if (e.which === 40) {
-            this.calendar.calendarContainer.focus();
+            (
```
Comment (emyarod) [inline packages/components/src/components/date-picker/date-picker.js:146]: should we also call `e.preventDefault()` so the page doesn't scroll?

File: packages/react/src/components/DatePicker/DatePicker.js [hunk 2]
```diff
@@ -413,6 +417,15 @@ export default class DatePicker extends Component {
           this.cal.close();
         }
       });
+      this.toInputField.addEventListener('keydown', e => {
+        if (e.which === 40) {
```
Comment (emyarod) [inline packages/react/src/components/DatePicker/DatePicker.js]: should we include a refactor to use these [keyboard handlers](https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/internal/keyboard/keys.js)?

**Review by emyarod — APPROVED** [review]
looks good to me, just a couple comments

**Review by abbeyhrt — APPROVED** [review]
I do agree with @emyarod that it would be cool to use the keyboard handlers instead of matching on key code but still looks good without those changes!

File: packages/react/src/components/DatePicker/DatePicker.js (hunk 2 above)
Comment (tw15egan) [reply packages/react/src/components/DatePicker/DatePicker.js]: Might as well: 

```js
import { match, keys } from '../../internal/keyboard';

if (match(event, keys.ArrowDown))
```

**Review by tw15egan — APPROVED** [review]
LGTM - but agree with @emyarod and should add `e.preventDefault()` to prevent page scroll + refactor to use keyboard util

**Comment by asudoh** [comment]
Makes sense, updated as such.

**Pull request merged**